File: nas_spec/code_samples/Node/workflows@{workflowId}@actions@{workflowActionId}/put.js
import { Checkout } from 'checkout-node-sdk';

let cko = new Checkout('your_client_secret_here', {
	client: 'ack_XXXXXXXXXXXX',
	scope: ['gateway'], // array of scopes
	environment: 'sandbox', // or "production"
});

// Or if you use api keys:
// const cko = new Checkout('sk_sbox_XXX', { pk: 'pk_sbox_XXX'}});

try {
	let workflows = await cko.workflows.updateAction('wf_2i7z3lwdoe5uzmomm7yzrytqdy', 'wfa_5qxwp7stgcqufj63mkr42xyeqi', {
		type: 'webhook',
		url: 'https://example.com/updated',
	});
} catch (err) {
	console.log(err.name);
}

File: nas_spec/code_samples/Go/workflows@{workflowId}@actions@{workflowActionId}/put.go
// For more information please refer to https://github.com/checkout/checkout-sdk-go
import (
	"github.com/checkout/checkout-sdk-go"
	"github.com/checkout/checkout-sdk-go/configuration"
	"github.com/checkout/checkout-sdk-go/workflows/actions"
)

// API Keys
api, err := checkout.
			Builder().
			StaticKeys().
			WithSecretKey("secret_key").
			WithEnvironment(configuration.Sandbox()). // or Environment.PRODUCTION
			Build()
if err != nil {
	return nil, err
}

// OAuth
api, err := checkout.
			Builder().
			OAuth().
			WithClientCredentials("client_id", "client_secret").
			WithScopes([]string{configuration.Flow, configuration.FlowWorkflows}).
			WithEnvironment(configuration.Sandbox()). // or Environment.PRODUCTION
			Build()
if err != nil {
	return nil, err
}

request := actions.NewWebhookActionRequest()
request.Url = "https://docs.checkout.com/webhook"
request.Headers = map[string]string{}
request.Signature = &actions.WebhookSignature{
	Key:    "8V8x0dLK%*DNS8JJr",
	Method: "HMACSHA256",
}

response, err := api.WorkFlows.UpdateWorkflowAction("workflow_id", "action_id", request)
if err != nil {
	return nil, err
}

return response, nil
